test/Control/TimeoutSpec.hs

```diff
@@ -0,0 +1,71 @@
+module Control.TimeoutSpec (spec) where
+
+import Control.Concurrent (newEmptyMVar, newMVar, threadDelay)
+import Control.Effect.Diagnostics (
+  Diagnostics,
+  FailureBundle (FailureBundle),
+  SomeDiagnostic (SomeDiagnostic),
+  ToDiagnostic (renderDiagnostic),
+  errorBoundary,
+  renderFailureBundle,
+ )
+import Control.Effect.Lift (Has, Lift)
+import Control.Timeout (
+  Cancel,
+  Duration (MicroSeconds),
+  checkForCancel,
+  durationToMicro,
+  shouldCancelRightNow,
+  timeout',
+ )
+import Control.Timeout.Internal (Cancel (Cancel))
+import Data.Functor (void)
+import Prettyprinter (pretty)
+import Test.Effect (expectationFailure', it', shouldBe')
+import Test.Hspec (Spec, describe, it, shouldBe)
+
+tinyDuration :: Duration
+tinyDuration = MicroSeconds 200
```

Comment:
[optional] [nit] prefer var name `twoHundredMicroseconds`

Reply:
The amount of time is arbitrary, the only relevant fact is that it's small.  Being specific about the amount of time actually reduces the value of the variable name.  e.g.: This variable would serve the exact same purpose if it were 175 _us_.